### PR TITLE
Fixing compilation errors for gles3 for Mac OS X

### PR DESCRIPTION
**Pull request by BastiaanOlij**
This is so far the first round of changes to get the compilation errors fixed on Mac OS X. More to follow. It compiles but is still broken at this point in time. Some of the fixes are coming from earlier PRs that went into the glew3 branch and didn't make it into the master.

Also there is a temporary fix in here until this issue is fixed upstream:
https://github.com/Dav1dde/glad/issues/70

(hopefully resolved this weekend..)

### CODE REVIEW

File: platform/windows/os_windows.cpp [hunk 1]
```diff
@@ -1088,14 +1088,14 @@ void OS_Windows::initialize(const VideoMode& p_desired,int p_video_driver,int p_
 	RasterizerGLES3::make_current();
 #else
 	// FIXME: Does DX support still work now that rasterizer is no longer used?
- #ifdef DX9_ENABLED
-	rasterizer = memnew( RasterizerDX9(hWnd) );
- #endif
+	#ifdef DX9_ENABLED
```
Comment (akien-mga) [inline platform/windows/os_windows.cpp]: Actually it was correct previously, we keep `#ifdef` unintended, and the rest of the code within such conditionals should keep its natural indentation (one tab here).

When you fix it, could you squash the commits into the first one? :)

Comment (BastiaanOlij) [inline platform/windows/os_windows.cpp]: You mean update the existing commits? sure but can I retrospectively do that?

Comment (akien-mga) [inline platform/windows/os_windows.cpp]: Yes, with `git commit --amend` to update the last commit (HEAD), or `git rebase -i HEAD~3` to rebase the last 3 commits interactively, which lets you define if you want to drop, edit or squash commits.

**Comment by BastiaanOlij** [comment]
Ok, we kinda have a working OSX build now. It suffers from the "screen is white until you resize" issue Reduz mentioned was also happening on windows and I've noticed that button clicks are sometimes being ignored. But we've got a starting point!

File: drivers/gles3/rasterizer_gles3.cpp [hunk 2]
```diff
@@ -136,9 +136,13 @@ void RasterizerGLES3::initialize() {
 		ERR_PRINT("Error initializing GLAD");
 	}
 
+#ifdef __APPLE__
```
Comment (akien-mga) [inline drivers/gles3/rasterizer_gles3.cpp]: This `#ifdef` could now go away, no?

Comment (andreasschuele) [inline drivers/gles3/rasterizer_gles3.cpp]: Maybe you are missing the GL_ARB_debug_output extension.

**Pull request merged**